### PR TITLE
Move Replay from KV to LF land

### LINKED ISSUE #13169: Splitting KV Replay
KV Replay is a very handy tool to quickly benchmark the change in the engine against big DAML projects that use kv ledger.
As the tool is depending on KV export decoding, it will have to move out the main repo with KV. 

Before completing the migration, we plan to split the tool in part (two parts leaving in LF and one  part leaving in KV).  The objectify of this split is double:
- on the one side it will simplify maintenance, as it will make API between KV and LV mroe clear
- on the other side, it will made the use of the tool handy, as it will make possible to benchmark the engine without direct dependency between KV and LF.

The API will relies  `SubmissionEntry` (which roughly correspond to TRANSACTION_ENTRY and PACKAGE_UPLOAD_ENTRY KV submission) and defined in proto as: 

```

message TransactionEntry {
    bytes rawTransaction = 1;
    string participantId = 2;
    repeated string submitters = 3;
    int64 ledgerTime = 4;
    int64 submissionTime = 5;
    bytes submissionSeed = 6;
}

message SubmissionEntry {
  oneof entry { 
    TransactionEntry transaction = 1;  
    bytes archives = 2;
  }
}
```

1.  The Export translator

The first part will be a KV tools that takes into input a KV ledger export and produce a stream of `SubmissionEntry`

2. TransactionSnapshot extractor

The second is a LF tools that take into input a stream of  `SubmissionEntry` and create a `TransactionSnapshot` 
 a TransactionSnapshot is the minimal set of information needed by the engine to replay a given transaction. 

3. The TransactionSnapshot replay

The third part is a LF binary tool that 
* take into input the file written by the extractor and benchmark the replay of the underlying transaction.

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
+ base it on submission entries instead of ledger export

fixes #13169

CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Review by cocreature — APPROVED** [review]
Thanks!

**Comment by remyhaemmerle-da** [comment]
@hubert-da Can you stamp the PR ?

File: ledger/participant-state/kvutils/tools/README.md [hunk 1]
```diff
@@ -34,53 +34,28 @@ The following options are supported:
 
 # `benchmark-replay`
```
Comment (hubert-da) [inline ledger/participant-state/kvutils/tools/README.md:35]: Let's just maybe rename this one for clarity.

**Review by hubert-da — APPROVED** [review]
Thank you vm and sorry for the delay! Just one minor comment.

File: ledger/participant-state/kvutils/tools/README.md (hunk 1 above)
Comment (remyhaemmerle-da) [reply ledger/participant-state/kvutils/tools/README.md:35]: Indeed !

**Review by miklos-da — APPROVED** [review]

File: daml-lf/snapshot/src/main/scala/com/daml/TransactionSnapshot.scala [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+    var idx: Int = index
+    var activeCreates = Map.empty[ContractId, Node.Create]
+    var archives = List.empty[ByteString]
+    var result = Option.empty[TransactionSnapshot]
+
+    try {
+
+      while (result.isEmpty && entries.hasNext) {
+        val entry = entries.next()
+        entry.getEntryCase match {
```
[the remark this answers is not in the document]
Comment (remyhaemmerle-da) [reply daml-lf/snapshot/src/main/scala/com/daml/TransactionSnapshot.scala:201]: 19ded43c926c92f615f346702c4f95e285926a31 is a first step toward the direction you advice.

**Pull request merged**